File: DSA/SakshamDwivedi_590013147/Day17/D17P1.c
#include <stdio.h>


int arr[100];
int tar;
int size;
int found = 0;
int main(){

    printf("Enter the size of the array (Less than 100): ");
    scanf("%d", &size);
    printf("Enter the elements (In sorted order) : ");
    for(int i = 0; i < size; i++){
        scanf("%d", &arr[i]);
    }
    printf("Enter the target value to search : ");
    scanf("%d", &tar);
    int start = 0;
    int end = size - 1;
    while(start <= end){
        int mid = start + (end - start) / 2;
        if(arr[mid] > tar){
            end = mid - 1;
        }
        else if(arr[mid] < tar){
            start = mid + 1;
        }
        else{
            printf("%d", mid);
            found = 1;
            break;
        }
    }

    if(!found){
        printf("-1");
    }
}

File: DSA/SakshamDwivedi_590013147/Day17/D17P1.py
#Solution Function
def binary_search(arr, target):
    start = 0
    end = len(arr) - 1
    while start <= end:
        mid = start + (end - start) // 2
        if arr[mid] > target:
            end = mid - 1
        elif arr[mid] < target:
            start = mid + 1
        else:
            return mid
    return -1

#Fixed Input
array = [1, 3, 5, 7, 9, 11]
tar = 7
print(binary_search(array, tar))

#Custom Input
array = list(int(x) for x in input("Enter an array (In sorted order) : ").split())
tar = int(input("Enter the target value : "))
print(binary_search(array, tar))
